SDWebImage/SDWebImageDownloader.m

```diff
@@ -135,7 +135,10 @@ - (void)setOperationClass:(Class)operationClass {
                                                          progress:^(NSInteger receivedSize, NSInteger expectedSize) {
                                                              SDWebImageDownloader *sself = wself;
                                                              if (!sself) return;
-                                                             NSArray *callbacksForURL = [sself callbacksForURL:url];
+                                                             __block NSArray *callbacksForURL;
+                                                             dispatch_sync(sself.barrierQueue, ^{
+                                                                 callbacksForURL = sself.URLCallbacks[url];
+                                                             });
```

Comment:
Not sure if this is fixes the problem . Taking pointer to NSArray using barrier doesn't solve threading problem: another thread can still modify array of callbacks while loop at rows  142-145 is using it.
Code:
callbacksForURL = [self.URLCallbacks[url] copy];
should help.